os/arch/arm/src/armv7-a/arm_cpuhotplug.c

```diff
@@ -83,24 +83,39 @@ int arm_hotplug_handler(int irq, void *context, void *arg)
 			   (ie. TizenRT Kernel logic is not ready to handle such case yet)
 			   Thus, we will be facing a deadlock when cpuA fire SGI2 to cpuB (ie. pausing CPU)
 	*/
+	int cpu = this_cpu();
+	/* If current cpu is in idle thread, then proceed for hotplug operation */
+	if (g_running_tasks[cpu]->pid == cpu) {
+		/* Set secondary core to hotplug state */
```

Comment:
Also i saw , g_running_tasks list is not getting updated anywhere in the code . if I am right , then i think we should use g_assignedtasks[cpu] to check

Reply:
Thank you for reminding~